metadata/src/main/java/org/apache/kafka/controller/FeatureControlManager.java

```diff
@@ -208,19 +218,6 @@ private ApiError updateMetadataVersion(
         boolean allowUnsafeDowngrade,
         Consumer<ApiMessageAndVersion> recordConsumer
     ) {
-        Optional<VersionRange> quorumSupported = quorumFeatures.quorumSupportedFeature(MetadataVersion.FEATURE_NAME);
-        if (!quorumSupported.isPresent()) {
-            return invalidMetadataVersion(newVersionLevel, "The quorum does not support metadata.version.");
-        }
-
-        if (newVersionLevel <= 0) {
```

Comment:
Should we allow a `metadata.version` of zero? Seems like it might complicate things to support a non-present metadata.version at runtime. Would that mean we need to revert back to reading the IBP from config?

Reply:
I was thinking about this too.

I would like to get rid of `MetadataVersion.UNINITIALIZED` as a concept. There are a few places where the MV isn't stored and it should be (pre-3.3 snapshots, for example) but we can just treat those as `MetadataVersion.3_3_0IV1`

That would allow us to just say that MV for a kraft cluster was always > 0

But this PR was already getting too big so I held off on that.